Make error properties non-enumerable

Weakky opened the pull request:
Closes #29

Weakky commented on test.js:
```diff
@@ -10,10 +10,9 @@ function deserializeNonError(t, value) {
-	t.true(keys.includes('name'));
-	t.true(keys.includes('stack'));
-	t.true(keys.includes('message'));
+	t.truthy(serialized.name);
+	t.truthy(serialized.stack);
+	t.truthy(serialized.message);
```
Not sure if this is good enough

sindresorhus replied:
I would do `t.true('message' in serialized);`

sindresorhus commented on index.js:
```diff
@@ -3,7 +3,10 @@
-		this.name = 'NonError';
+		Object.defineProperty(this, 'name', {
+			value: 'NonError',
+			enumerable: false
```
This is the default.

sindresorhus replied:
However, it should be:

```
{value: "Error", writable: true, enumerable: false, configurable: true}
configurable: true
enumerable: false
value: "Error"
writable: true
__proto__: Object
```

sindresorhus commented on test.js:
```diff
@@ -172,3 +171,17 @@ test('should deserialize plain object', t => {
+	const deserialized = deserializeError(object);
+	const enumerableProps = Object.keys(deserialized);
+
+	t.true(enumerableProps.length === 0);
```
Use `t.is`

sindresorhus commented on test.js:
```diff
@@ -172,3 +171,17 @@ test('should deserialize plain object', t => {
+		message: 'error message',
+		stack: 'at <anonymous>:1:13',
+		name: 'name',
+		code: 'code'
```
`code` is enumerable, at least in Node.js:

```js
> try {
... 	fs.readFileSync('sadasd')
... } catch (err) {
... 	console.log(Object.keys(err));
... }
[ 'errno', 'syscall', 'code', 'path' ]
```

Weakky replied:
Ah, good catch!

sindresorhus commented:
Can you explicitly mention in the readme which properties are enumerable and which are not?

Weakky commented:
Review done, I made `code` enumerable and adjusted the test so that it also makes sure that enumerable props are kept enumerable.

Readme + TS docs are also updated
